backend/src/authentication/authentication.service.ts

```diff
@@ -18,23 +18,25 @@ export class AuthenticationService {
         private readonly jwtService: JwtService) { }
 
     async loginAdmin(loginData: LoginAdminDto): Promise<JWTToken> {
-        const user = await this.adminService.getAdmin(loginData.email);
+        const user = await this.adminService.getAdminByEmail(loginData.email);
         if (!user) { throw new BadRequestException(content.UserNotFound); }
         return await this.validateUser({
-            providedPassword: loginData.password, hashedPassword: user.password,
+            provided: loginData.password, expected: user.password,
         }, user.id);
     }
 
     async loginJunior(loginData: LoginJuniorDto): Promise<JWTToken> {
-        const user = await this.juniorService.getJunior(loginData.phoneNumber);
+        const user = await this.juniorService.getJunior(loginData.id);
         if (!user) { throw new BadRequestException(content.UserNotFound); }
         return await this.validateUser({
-            providedPassword: loginData.pin, hashedPassword: user.pin,
+            provided: loginData.challenge,
         }, user.id);
     }
 
-    async validateUser(attempt: { providedPassword: string, hashedPassword: string }, user: string): Promise<JWTToken> {
-        const passwordMatch = await compare(attempt.providedPassword, attempt.hashedPassword);
+    async validateUser(attempt: { provided: string, expected?: string }, user: string): Promise<JWTToken> {
```

Comment:
Is there a reason for the attempt to be it own object, would passing just 2 parameters suffice? In my opinion that would be a bit clearer. Would also prefer if the third paramater `user` would be named something like userId as (atleast for me) just having `user` implies it's an object. It's typescript so you can see the type, I know, but just something I'd personally prefer. I accept differing opinions on this. 😄

Reply:
As the attempt is always provided, but the expected is optional, I felt the object felt cleaner than having say "user, providedAttempt, expected?".
With regards to the naming, I had not considered that. I agree and shall change that.

Reply:
I personally dont see the problem of having an optional argument / having a default value. In my opinion its nicer than having to always pass an object. If you have strong feelings about this, we can go with your solution however.